src/lib/agent/LedgerService.ts

```diff
@@ -0,0 +1,181 @@
+import logger from '../logger';
+import { Wallet } from '../wallet/Wallet';
+
+export class LedgerService {
+  wallet: Wallet;
+  indy: Indy;
+  poolHandle?: PoolHandle;
+  authorAgreement?: AuthorAgreement;
+
+  constructor(wallet: Wallet, indy: Indy) {
+    this.wallet = wallet;
+    this.indy = indy;
+  }
+
+  async connect(poolName: string, poolConfig: PoolConfig) {
+    try {
+      logger.log(__dirname);
+
+      logger.log('Creating pool config');
+      await this.indy.createPoolLedgerConfig(poolName, poolConfig);
+    } catch (e) {
+      logger.log('PoolLedgerConfigAlreadyExists');
```

Comment:
This should be below the line that checks whether it is indeed a `PoolLedgerConfigAlreadyExistsError`